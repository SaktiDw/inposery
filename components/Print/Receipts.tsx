import axiosInstance from "@/helper/lib/client";
import React, { forwardRef, useEffect, useState } from "react";
import useSWR from "swr";
import PriceFormater from "../FormatText/PriceFormater";

export const Receipts = forwardRef((props: any, ref: any) => {
  if (!props.data) return <>Loading</>;
  return (
    <div ref={ref} className="flex flex-col rounded-lg p-4 w-[300px] text-sm">
      {/* <header className="text-xl font-bold">Toko Inposery</header>
      <table className="table table-auto border ">
        <thead>
          <tr>
            <td>No</td>
            <td>Name</td>
            <td>Qty</td>
            <td>Harga</td>
          </tr>
        </thead>
        <tbody>
          {props.data &&
            JSON.parse(props.data.product).map((item: any, index: number) => {
              return (
                <tr className="border">
                  <td className="p-2">{index + 1}</td>
                  <td className="p-2">{item.name}</td>
                  <td className="p-2">{item.qty}</td>
                  <td className="p-2">
                    <PriceFormater price={item.price} />
                  </td>
                </tr>
              );
            })}
        </tbody>
      </table>

      <footer>
        <div className="flex item-center justify-between">
          <span>subTotal</span>
          <PriceFormater price={props.data.total} />
        </div>
        <div className="flex item-center justify-between">
          <span>payment</span>
          <PriceFormater price={props.data.payment} />
        </div>
        <div className="flex item-center justify-between">
          <span>change</span>
          <PriceFormater price={props.data.change} />
        </div>
      </footer> */}
      <div className="rounded-lg shadow-lg p-12 text-xs mt-8 mx-4 sm:mx-0 bg-white">
        <div>
          <div className="flex flex-col">
            <h1 className="text-gray-800 text-xl font-medium mb-2">Receipt</h1>
            <p className="text-gray-600 text-xs">
              Date: {props.data.createdAt}
            </p>
            <p className="text-gray-600 text-xs">
              Order Number: {props.data.id}
            </p>
          </div>
          <hr className="my-4" />
          <div>
            {props.data &&
              JSON.parse(props.data.product).map((item: any, index: number) => {
                return (
                  <div key={index}>
                    <div className="flex justify-between items-center">
                      <span className="font-medium text-base">{item.name}</span>
                      <PriceFormater
                        price={item.sellPrice}
                        className="text-base font-medium"
                      />
                    </div>
                    <div className="mb-4 flex justify-between items-center">
                      <span>*Quantity:</span>
                      <span className="">{item.qty}</span>
                    </div>
                  </div>
                );
              })}

            <div className="flex justify-between items-center">
              <span className="text-lg font-medium">Total</span>
              <PriceFormater
                price={parseInt(props.data.total)}
                className="text-lg font-medium"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
});

Receipts.displayName = "Receipts";
export default Receipts;
