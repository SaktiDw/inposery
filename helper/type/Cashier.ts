export interface Cart {
  id: number;
  name: string;
  sell_price: number;
  qty: number;
  orderQty: number;
  customer: Customer;
}

export interface Customer {
  name: string;
  active: boolean;
}
