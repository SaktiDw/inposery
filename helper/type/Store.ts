export interface Store {
  id: number;
  name: string;
  user_id: number;
  created_at: Date;
  updated_at: Date;
  media?: any[];
}

export interface Link {
  url: string;
  label: string;
  active: boolean;
}

export interface StoresResponse {
  current_page: number;
  data: Store[];
  first_page_url: string;
  from: number;
  last_page: number;
  last_page_url: string;
  links: Link[];
  next_page_url?: any;
  path: string;
  per_page: number;
  prev_page_url?: any;
  to: number;
  total: number;
}

export interface StoreInput {
  name: string;
  user: number;
  image: string;
}
