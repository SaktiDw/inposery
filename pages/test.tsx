import {
  Pagination,
  PerPageSelect,
  SearchInput,
  StoreDashboard,
  Table,
} from "@/components/index";
import { TableColumn } from "@/components/Tables/Table";

import axios from "@/helper/lib/api";
import { useRouter } from "next/router";
import React, { useState } from "react";
import useSWR from "swr";
import qs from "qs";

type Props = {};

type Selected = {
  selected: boolean;
  id: number;
};

const Transaction = (props: Props) => {
  const router = useRouter();
  const storeId = router.query.id;

  const [perPage, setPerPage] = useState("1000");
  const [search, setSearch] = useState("");
  const [pageIndex, setPageIndex] = useState(1);

  const query = qs.stringify(
    {
      filter: { "product.name": `${search}`, store_id: storeId },
      limit: perPage,
      page: pageIndex,
    },
    { encodeValuesOnly: true }
  );
  const {
    data: products,
    error,
    mutate,
  } = useSWR(`/api/test?${query}`, (url) =>
    axios.get(url).then((res) => res.data)
  );
  const [selected, setSelected] = useState<Selected[]>([]);

  const columns: TableColumn<any>[] = [
    { title: "#", key: "id", dataType: "numbering" },
    {
      title: "Name",
      key: "product",
      render(val, index) {
        return val.product?.name;
      },
    },
    {
      title: "Price",
      key: "price",
      render: (val) => (
        <div className="flex gap-4">
          {val.price.toLocaleString("id-ID", {
            style: "currency",
            currency: "IDR",
          })}
        </div>
      ),
    },
    { title: "Quantity", key: "qty" },
    { title: "Type", key: "type" },
    {
      title: "Date",
      key: "created_at",
      render: (val) => new Date(val.created_at).toLocaleString(),
    },
  ];

  return (
    <StoreDashboard>
      <div className="flex flex-col gap-4">
        <div className="flex gap-4">
          <PerPageSelect onChange={(e) => setPerPage(e.target.value)} />
          <SearchInput onChange={(e) => setSearch(e.target.value)} />
        </div>
        {products && (
          <>
            <Table data={products} columns={columns} />
          </>
        )}
      </div>
      {products && <Pagination data={products} setPage={setPageIndex} />}
    </StoreDashboard>
  );
};

export default Transaction;
