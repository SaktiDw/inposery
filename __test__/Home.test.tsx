import { mockNextUseRouter } from "@/helper/test/utils";
import MainDashboard from "@/pages/dashboard";
import Login from "@/pages/login";
import Stores from "@/pages/store";
import Store from "@/pages/store/[id]";
import Cashier from "@/pages/store/[id]/cashier";
import Products from "@/pages/store/[id]/product";
import Receipts from "@/pages/store/[id]/receipts";
import Transaction from "@/pages/store/[id]/transaction";
import { render } from "@testing-library/react";
import { describe, expect, it, vi, vitest } from "vitest";
import Home from "../pages";

describe("Page Render Testing", () => {
  vitest.mock("next/router", () => ({
    useRouter: () => ({
      query: { id: "1" },
    }),
  }));

  it("Should be render Home Page", () => {
    const { container } = render(<Home />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
  it("Should be render Login Page", () => {
    const { container } = render(<Login />);
    expect(container).toBeDefined();
  });
  it("Should be render Dashboard Page", () => {
    const { container } = render(<MainDashboard />);
    expect(container).toBeDefined();
  });
  it("Should be render Stores Page", () => {
    const { container } = render(<Stores />);
    expect(container).toBeDefined();
  });
  it("Should be render Store Page", () => {
    const { container } = render(<Store />);
    expect(container).toBeDefined();
  });
  it("Should be render Products Page", () => {
    const { container } = render(<Products />);
    expect(container).toBeDefined();
  });
  it("Should be render Cashier Page", () => {
    const { container } = render(<Cashier />);
    expect(container).toBeDefined();
  });
  it("Should be render Transaction Page", () => {
    const { container } = render(<Transaction />);
    expect(container).toBeDefined();
  });
  it("Should be render Receipts Page", () => {
    const { container } = render(<Receipts />);
    expect(container).toBeDefined();
  });
});
