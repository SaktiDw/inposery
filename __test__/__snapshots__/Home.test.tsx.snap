// Vitest Snapshot v1

exports[`Page Render Testing > Should be render Home Page 1`] = `
<div>
  <main
    class="w-full h-screen flex items-center justify-center bg-slate-100 dark:bg-slate-900 text-slate-800 dark:text-white"
  >
    <nav
      class="fixed top-0 left-0 z-50 w-full px-6 py-3 flex items-center justify-between bg-slate-100 bg-opacity-50 backdrop-blur dark:bg-slate-900 dark:backdrop-blur-lg dark:bg-opacity-50"
    >
      <div>
        <button
          class="p-3"
        >
          <i
            class="fi-rr-menu-burger"
          />
        </button>
        <span
          class="font-bold text-2xl"
        >
          In
          <span
            class="text-lime-500"
          >
            POS
          </span>
          ery
        </span>
      </div>
      <div
        class="flex gap-2 "
      >
        <a
          href="/login"
        >
          Login
        </a>
        <a
          href="/register"
        >
          Register
        </a>
      </div>
    </nav>
    <h1
      class="text-slate-800 dark:text-white text-3xl"
    >
      Landing Page â™¥
    </h1>
    landing-page
    <a
      href="/dashboard"
    >
      Get Started
    </a>
  </main>
</div>
`;
